cmd/kubeadm/app/master/selfhosted_volumes.go

```diff
@@ -0,0 +1,294 @@
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createTLSSecrets(client *clientset.Clientset) error {
+	tlsPairs := []string{
+		kubeadmconstants.CACertAndKeyBaseName,
+		kubeadmconstants.CACertName,
+		kubeadmconstants.CAKeyName,
+		kubeadmconstants.APIServerCertAndKeyBaseName,
+		kubeadmconstants.APIServerCertName,
+		kubeadmconstants.APIServerKeyName,
+		kubeadmconstants.APIServerKubeletClientCertAndKeyBaseName,
+		kubeadmconstants.APIServerKubeletClientCertName,
+		kubeadmconstants.APIServerKubeletClientKeyName,
+		kubeadmconstants.ServiceAccountKeyBaseName,
+		kubeadmconstants.ServiceAccountPublicKeyName,
+		kubeadmconstants.ServiceAccountPrivateKeyName,
+		kubeadmconstants.FrontProxyCACertAndKeyBaseName,
+		kubeadmconstants.FrontProxyCACertName,
+		kubeadmconstants.FrontProxyCAKeyName,
+	}
+	for i := 0; i < len(tlsPairs); i = i + 3 {
+		fmt.Printf("[self-hosted] Creating TLS secret %q with %s and %s\n", tlsPairs[i], tlsPairs[i+1], tlsPairs[i+2])
+		secret, err := createTLSSecretFromFiles(
+			tlsPairs[i],
+			path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", tlsPairs[i+1]),
+			path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", tlsPairs[i+2]),
+		)
+		if err != nil {
+			return err
+		}
+
+		if _, err := client.Secrets(metav1.NamespaceSystem).Create(secret); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createOpaqueSecrets(client *clientset.Clientset) error {
+	files := []string{
+		kubeadmconstants.SchedulerKubeConfigFileName,
+		kubeadmconstants.ControllerManagerKubeConfigFileName,
+	}
+	for i := 0; i < len(files); i++ {
+		fmt.Printf("[self-hosted] Creating secret for %q\n", files[i])
+		secret, err := createOpaqueSecretFromFile(
+			files[i],
+			path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, files[i]),
+		)
+		if err != nil {
+			return err
+		}
+
+		if _, err := client.Secrets(metav1.NamespaceSystem).Create(secret); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createTLSSecretFromFiles(secretName, crt, key string) (*v1.Secret, error) {
+	data := make(map[string][]byte, 0)
+
+	crtBytes, err := ioutil.ReadFile(crt)
+	if err != nil {
+		return nil, err
+	}
+	data["tls.crt"] = crtBytes
+
+	keyBytes, err := ioutil.ReadFile(key)
+	if err != nil {
+		return nil, err
+	}
+	data["tls.key"] = keyBytes
+
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: metav1.NamespaceSystem,
+		},
+		Type: v1.SecretTypeTLS,
+		Data: data,
+	}, nil
+}
+
+func createOpaqueSecretFromFile(secretName, file string) (*v1.Secret, error) {
+	data := make(map[string][]byte, 0)
+
+	fileBytes, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+	data[path.Base(file)] = fileBytes
+
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: metav1.NamespaceSystem,
+		},
+		Type: v1.SecretTypeOpaque,
+		Data: data,
```

Comment:
suggest:
```go
Data: map[string][]byte{
   path.Base(file): fileBytes,
}
```
to avoid creating the extra var when it's not consumed anywhere else